File: backend/backend/app/schemas/meeting.py
from pydantic import BaseModel, EmailStr
from typing import List, Optional
from datetime import datetime

class AttendeeBase(BaseModel):
    email: str
    name: Optional[str] = None

class AttendeeCreate(AttendeeBase):
    pass

class Attendee(AttendeeBase):
    id: str
    meeting_id: str

    class Config:
        from_attributes = True

class MeetingBase(BaseModel):
    audio_url: str

class MeetingCreate(MeetingBase):
    pass

class Meeting(MeetingBase):
    id: str
    user_id: str
    transcript: Optional[str] = None
    summary: Optional[str] = None
    status: str
    duration_seconds: Optional[int] = None
    created_at: datetime
    attendees: List[Attendee] = []

    class Config:
        from_attributes = True

class ContactBase(BaseModel):
    email: EmailStr
    name: Optional[str] = None

class ContactCreate(ContactBase):
    pass

class Contact(ContactBase):
    id: str
    user_id: str
    created_at: datetime

    class Config:
        from_attributes = True

File: donsnotes-android/app/src/main/java/com/donsnotes/app/data/model/Meeting.kt
package com.donsnotes.app.data.model

import com.google.gson.annotations.SerializedName

data class Meeting(
    @SerializedName("id") val id: String,
    @SerializedName("status") val status: MeetingStatus,
    @SerializedName("audio_url") val audioUrl: String? = null,
    @SerializedName("transcript") val transcript: String? = null,
    @SerializedName("summary") val summary: String? = null,
    @SerializedName("attendees") val attendees: List<Attendee> = emptyList(),
    @SerializedName("organizer_name") val organizerName: String? = null,
    @SerializedName("created_at") val createdAt: String? = null,
    @SerializedName("user_id") val userId: String? = null
)
